packages/walletconnect/src/index.ts

```diff
@@ -73,47 +67,57 @@ export class WalletConnect extends Connector {
     if (error) this.onError?.(error)
   }
 
-  private chainChangedListener = (chainId: number | string): void => {
-    this.actions.update({ chainId: parseChainId(chainId) })
+  private chainChangedListener = (chainId: number): void => {
+    this.actions.update({ chainId })
   }
 
   private accountsChangedListener = (accounts: string[]): void => {
     this.actions.update({ accounts })
   }
 
-  private URIListener = (_: Error | null, payload: { params: string[] }): void => {
-    this.events.emit(URI_AVAILABLE, payload.params[0])
+  private URIListener = (uri: string): void => {
+    this.events.emit(URI_AVAILABLE, uri)
   }
 
-  private async isomorphicInitialize(chainId = this.defaultChainId): Promise<void> {
-    if (this.eagerConnection) return
-
-    // because we can only use 1 url per chainId, we need to decide between multiple, where necessary
-    const rpc = Promise.all(
-      Object.keys(this.rpc).map(
-        async (chainId): Promise<[number, string]> => [
-          Number(chainId),
-          await getBestUrl(this.rpc[Number(chainId)], this.timeout),
-        ]
-      )
-    ).then((results) =>
-      results.reduce<{ [chainId: number]: string }>((accumulator, [chainId, url]) => {
-        accumulator[chainId] = url
-        return accumulator
-      }, {})
-    )
-
-    return (this.eagerConnection = import('@walletconnect/ethereum-provider').then(async (m) => {
-      this.provider = new m.default({
+  private async isomorphicInitialize(desiredChainId: number | undefined = this.defaultChainId): Promise<MockWalletConnectProvider> {
+    if (this.eagerConnection) return this.eagerConnection
+
+    if (this.provider) {
+      await this.deactivate()
+    }
+
+    const rpcMap = this.rpcMap
```

Comment:
nit: Can this transform be moved into a private helper method to declutter `isomorphicInitialize`? I'd be ok saving the passed `rpcMap` to `this._rpcMap` and making `this.rpcMap` a private getter that does this processing.

Reply:
Yes, we can do it for to keep it clean. Also, while we're at it, I don't think we need to cache the results, now that we're not really deactivating/activating provider anyway, I think this computation runs just once anyway.

Reply:
There're no async getters/setters, but I'll find another (clean) way

Reply:
I decided to move it to a separate utility file and as a result, we have increased coverage. I felt like that Promise.all logic was part of utility too. I am not 100% sold on accessing object like array (`rpc[0]` instead of rpc['0'] but it works thanks to JS (tm) and saves few keystrokes). Open to discussion!